src/tmpfiles/tmpfiles.c

```diff
@@ -1370,7 +1370,7 @@ static int truncate_file(Item *i, const char *path) {
 
         assert(i);
         assert(path);
-        assert(i->type == TRUNCATE_FILE);
+        assert(i->type == TRUNCATE_FILE || i->type == CREATE_FILE);
```

Comment:
Should this be: `assert(i->type == TRUNCATE_FILE || (i->type == CREATE_FILE && i->append_or_force))` ?

Reply:
I think so.